internal/controller/streams.go

```diff
@@ -197,105 +167,289 @@ func (sm *streamManager) BootUpStreamAndReconnect(ctx context.Context, cancel co
+					close(done)
+					return
+				}
+
+				randomInt, err := rand.Int(rand.Reader, max)
+				if err != nil {
+					logger.Errorw("Could not generate a random int", "error", err)
+					sleepTimer.Reset(backoff) // Reset sleep timer before continuing loop
+					continue
+				}
+
+				jitterPct, _ := randomInt.Float64()
+				jitterPct = jitterPct * maxJitterPct
+				sleep := time.Duration(float64(backoff) * (1. - jitterPct))
+
+				if sleep < initialBackoff {
+					sleep = initialBackoff
+				}
+				if sleep > maxBackoff {
+					sleep = maxBackoff
+				}
+
+				logger.Infow("Sleeping before retrying connection", "backoff", sleep)
+				sleepTimer.Reset(sleep) // Reset sleep timer with calculated backoff delay
+
+				backoff *= 2
+				if backoff > maxBackoff {
+					backoff = maxBackoff
+				}
+			} else {
+				consecutiveFailures = 0
+				backoff = initialBackoff
+				// Reset sleep timer back to initial backoff interval
+				sleepTimer.Reset(initialBackoff)
 			}
-			responseData, err := Onboard(ctx, envMap.TlsSkipVerify, envMap.OnboardingEndpoint, OnboardingCredentials, logger)
+		}
+	}
+}
+
+// ConnectStreams will continue to reboot and restart the main operations within the operator if any disconnects or errors occur.
+func ConnectStreams(ctx context.Context, logger *zap.SugaredLogger, envMap EnvironmentConfig, bufferedGrpcSyncer *BufferedGrpcWriteSyncer) {
+	// Timer channel for 5 seconds
+	timer := time.After(5 * time.Second)
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-timer:
+			authConn, client, err := NewAuthenticatedCon(ctx, logger, envMap)
 			if err != nil {
-				logger.Errorw("Failed to register cluster", "error", err)
+				logger.Errorw("Failed to establish initial connection; will retry", "error", err)
 				continue
 			}
-			err = authn.WriteK8sSecret(ctx, responseData, envMap.ClusterCreds)
-			time.Sleep(1 * time.Second)
-			if err != nil {
-				logger.Errorw("Failed to write secret to Kubernetes", "error", err)
+
+			streamClient := &streamClient{
+				conn:            authConn,
+				client:          client,
+				ciliumNamespace: envMap.CiliumNamespace,
 			}
-			clientID, clientSecret, err = authn.ReadCredentialsK8sSecrets(ctx, envMap.ClusterCreds)
-			if err != nil {
-				logger.Errorw("Could not read K8s credentials", "error", err)
-				continue
+
+			sm := &streamManager{
+				streamClient:       streamClient,
+				logger:             logger,
+				bufferedGrpcSyncer: bufferedGrpcSyncer,
 			}
+
+			resourceDone := make(chan struct{})
+			logDone := make(chan struct{})
+			ciliumDone := make(chan struct{})
+			sm.bufferedGrpcSyncer.done = logDone
+
+			go manageStream(logger, connectAndStreamResources, sm, resourceDone)
+			go manageStream(logger, connectAndStreamLogs, sm, logDone)
+			// Only start network flows stream if not disabled
+			if !sm.streamClient.disableNetworkFlows {
+				go manageStream(logger, connectAndStreamCiliumNetworkFlows, sm, ciliumDone)
```

Comment:
Only create the channel if you need it.
```suggestion
				ciliumDone = make(chan struct{})
				go manageStream(logger, connectAndStreamCiliumNetworkFlows, sm, ciliumDone)
```